store/file/ods.go

```diff
@@ -368,17 +375,25 @@ func readRow(fl io.ReaderAt, hdr *headerV0, sharesOffset, quadrantIdx, rowIdx in
 	offset := sharesOffset + quadrantOffset + rowOffset
 
 	axsData := make([]byte, odsLn*shrLn)
-	if _, err := fl.ReadAt(axsData, int64(offset)); err != nil {
+	n, err := fl.ReadAt(axsData, int64(offset))
+	if err != nil && !errors.Is(err, io.EOF) {
```

Comment:
Should we just `return nil, nil` on `io.EOF` ?

Reply:
Continuing on EOF is intentional and the core of this change. If it reads beyond the file boundaries it feels the row shares with TailPadding shares.

Reply:
So the write side ends writing earlier if it sees tail padding shares, and the read side adds it back if the file is less than the expected data.